ext/spl/spl_observer.stub.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+interface SplObserver
+{
+    /** @return void */
+    public function update(SplSubject $subject);
+}
+
+interface SplSubject
+{
+    /** @return void */
+    public function attach(SplObserver $observer);
+
+    /** @return void */
+    public function detach(SplObserver $observer);
+
+    /** @return void */
+    public function notify();
+}
+
+class SplObjectStorage implements Countable, Iterator, Serializable, ArrayAccess
+{
+    /**
+     * @param mixed $info
+     * @return void
+     */
+    public function attach(object $object, $info = null) {}
+
+    /** @return void */
+    public function detach(object $object) {}
+
+    /** @return bool */
+    public function contains(object $object) {}
+
+    /** @return int */
+    public function addAll(SplObjectStorage $storage) {}
+
+    /** @return void */
+    public function removeAll(SplObjectStorage $storage) {}
+
+    /** @return void */
+    public function removeAllExcept(SplObjectStorage $storage) {}
+
+    /** @return mixed */
+    public function getInfo() {}
+
+    /**
+     * @param mixed $info
+     * @return void
+     */
+    public function setInfo($info) {}
+
+    /** @return int */
+    public function count(int $mode = COUNT_NORMAL) {}
+
+    /** @return void */
+    public function rewind() {}
+
+    /** @return bool */
+    public function valid() {}
+
+    /** @return int */
```

Comment:
Ugh, typical SPL. This whole class would make so much more sense if this was returning the object, current() the data and getInfo/setInfo were gone...